File: reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        """ 
        1) Convert absolute value of x to a string
        2) Reverse the string via slicing
        3) Convert reversed string back to integer
        4) Return 0 if out of bounds, else determine sign and return result
        """
        res = int(str(abs(x)) [::-1])            
        if res >= (2**31) - 1 or res <= -(2**31):
            return 0
        return res if x > 0 else -res

File: reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        
        if (x >= INT_MAX || x <= INT_MIN) return 0;
        
        ostringstream oss;
        oss << x;
        string num = oss.str();

        int res;
        int sign = x > 0 ? 1 : -1;
        int len = num.length();
        for (int i = 0; i < len/2; i++)
            swap(num[i], num[len-i-1]);

        try {
            res = stoi(num);
        }
        catch (out_of_range& e) {
            return 0; 
        }
        
        return sign * res;
    }
};